packages/app/source/client/controllers/todos-controller.js

```diff
@@ -6,13 +6,17 @@ Space.Object.extend(Todos, 'TodosController', {
   ],
 
   dependencies: {
-    configuration: 'configuration'
+    configuration: 'configuration',
+    todosStore: 'Todos.TodosStore',
```

Comment:
@rhyslbw: I was considering how much responsibility `TodosController` should handle when I was writing this:

```
_onTodoToggled(event) {
    if (event.isCompleted) {
      this.send(new Todos.ReopenTodo({
        targetId: this.configuration.todoListId,
        todoId: new Guid(event.id)
      }));
    } else {
      this.send(new Todos.CompleteTodo({
        targetId: this.configuration.todoListId,
        todoId: new Guid(event.id)
      }));
    }
  },
```
I think the situation is analogous to toggling all todos. 

But maybe you are right, component needs to check the store and send ui events that controller can translate to api commands without interacting with the store.

In Donations there are also dependancies to stores in controllers.